### DISCUSSION TITLE
AliSim codon models

### DISCUSSION THREAD

**Discussion by thednainus** [iqtree/iqtree2#196] [Q&A] [answered]
Hello,

I am trying to simulate some sequence alignment using a codon model.
I have tried the following command:

`iqtree2 --alisim alignment_codon -m MG2K{1.0,0.3,0.5} -t tree.nwk`
tree.nwk is the same tree as the example in here http://www.iqtree.org/doc/AliSim

I get the following error message:

_IQ-TREE multicore version 2.3.2 COVID-edition for Mac OS X 64-bit built Apr  9 2024
Developed by Bui Quang Minh, James Barbetti, Nguyen Lam Tung,
Olga Chernomor, Heiko Schmidt, Dominik Schrempf, Michael Woodhams, Ly Trong Nhan, Thomas Wong_

_Host:    Fabricias-Air.home (SSE4.2, 16 GB RAM)_
_Command: iqtree2 --alisim alignment_codon -m MG2K1.0 MG2K0.3 MG2K0.5 -t tree.nwk_
_Seed:    942363 (Using SPRNG - Scalable Parallel Random Number Generator)_
_Time:    Tue May 14 10:59:10 2024_
_OMP: Info #276: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead._
_Kernel:  SSE2 - 1 threads (8 CPU cores detected)_

_[Alignment Simulator] Executing_
**ERROR: File not found or incorrect input, pls check it again.MG2K1.0**

Do you know why this is happens? 
I am just trying to understand how AliSim works for codon models. 
I took this parameter value -m MG2K{1.0,0.3,0.5} from the example in here http://www.iqtree.org/doc/Substitution-Models
I use a Mac OS Sonoma M2 chip.

Also note that if I use this example here:
`iqtree2 --alisim alignment_codon -m MG{2.0}+F1X4{0.2/0.3/0.4/0.1} -t tree.nwk`
It works fine.

Thanks,
Fabricia.

**Comment by trongnhanuit** [comment, marked as answer]
Dear Fabricia,
Thank you for using AliSim. Can you try with:

iqtree2 --alisim alignment_codon -m "MG2K{1.0,0.3,0.5}" -t tree.nwk

(with "" covers the model and its parameters).

Cheers,
Nhan

**Reply by thednainus** [reply]
Hello!
Thanks for the replay! It does work in this way.
I guess I got confused because in the example that the model works, the "" are not needed.

Cheers,
Fabricia.

**Reply by bqminh** [reply]
The command line parser in Linux/Mac is doing something funny when you have a comma-separated string enclosed by curly brackets. If you compare your original command `-m MG2K{1.0,0.3,0.5}` with what iqtree printed in the log file (See "Command:" line), it becomes `-m MG2K1.0 MG2K0.3 MG2K0.5`. That's the reason.

Apart from the solution using single-quote or double-quote around the string to avoid that as @trongnhanuit suggested, you can also use slash-separated string, e.g., `-m MG2K{1.0/0.3/0.5}`. Slash is introduced as an alternative separator for comma. Then you don't need to use quotes.
